### ISSUE TITLE
Sequencer: Changing length of sequencer block: Start/stop playing inside the block, and automatic looping.

### ISSUE THREAD

**Issue by kmatheussen** [kmatheussen/radium#1049]
Copied from @Lydkraft (#741):

1. Block looping. Obviously. Would allow working with blocks that have
a small number of lines per block without the need to manually copy stuff around 1000 times.
(Common in standard sequencers, so expect people to expect this from Radium as well.
A temporary "infinite loop" would already immensely help in a lot of cases.)

Also note: With the ability to go full polyrhythm now (different block lengths/lines)
this would make it also easy to change length of blocks on the fly to find the best
"polyrhythm" block length without having to move all copied blocks around and lining them
up again against each other each time when the length is changed...)

2. Ability to shorten the block in the sequencer per dragging the endpoint.
Would allow making the essential pauses very easily on an arrangement. Something
like that would be very useful. (Also super-common in standard sequencers. Probably will also
be requested pretty quickly.)

**Comment by kmatheussen** [comment]
I guess these are important also because it feels like something you should be able to do, and people might get frustrated trying (and failing) to do it.

**Comment by Lykdraft** [comment]
Better/easier  idea for point 2: "shorten blocks in sequencer".
Use a "Mask" Handle. (That might just make the "masked part of the block darker and not really shorten it. The mask would then just mute all tracks of the block. Might be much easier to realize and
would have the same effect as shorten the actual block.

Best,
tL.

**Comment by kmatheussen** [comment]
You can already simulate changing start and end positions of seqblocks by
using seqblock volume automation. I think that's better than using track
mute.

The problem with changing start and end positions for editor seqblocks is
that there's lots of code in the scheduler that must be extended, which can
be quite hairy. It's something that will be done after V6.

**Comment by Lykdraft** [comment]
Okay I'll check this out. I hope you can snap the "Volume Automation" to subpositions on the grid, otherwise this probably isn't a good option at all. (i.e.: Way too fiddly to hit exactly 1/2 beat before start of next bar. )
Given the fact that  #937 is still open, I don't think that its easy to deal with this for exact off's.
But I'll test it out.

**Comment by kmatheussen** [comment]
Everything snaps to grid when grid is enabled.

#937 has actually been implemented for a very long time. I forgot
that there was a ticket for it. But right now it's not there.
It was disabled in 5.9.34, I think, since the whole grid painting was
redone.
But it should be put back soon.

**Comment by Lykdraft** [comment]
I see. By the way: Where is the "Seqblock Volume Automation"?

**Comment by kmatheussen** [comment]
Right click seqblock and enable the "Volume Automation" checkbox.

**Comment by Lykdraft** [comment]
Alright thanks.

**Comment by kmatheussen** [comment]
It's very convenient to use seqblock automation for this if you
deselect "Glide to next node" on the first node. Then you
only need two nodes. The first node you set at 0.0dB,
and the second node you set at -Inf dB.

**Comment by Lykdraft** [comment]
Oh yeah indeed. This is easy.

**Comment by kmatheussen** [comment]
No, sorry, volume before the first node is set to the value of the first
node.
So you need three nodes, where the first node is also non-gliding and set
to 0dB
Still very convenient though.

**Comment by Lykdraft** [comment]
To add to this:
Another idea would be to have the white "X"'s also available per-grid setting.
So that you could click on a part of the block (depending on grid size, or beat division)
and mute a part of the block. (with a white "X" showing where the block is muted.)

This would tie in pretty nicely with the design and also would eliminate the need to
tackle the option for custom block lenght.

Maybe worth a thought.

**Comment by yust1n** [comment]
Shorten a block can be done with Editor Edit/Block/Split. Just move to the place you want to split in the editor. Since there is no concat ( yet ? )  best would be to duplicate that block in the sequencer first.
